sklearn/feature_selection/_from_model.py

```diff
@@ -99,8 +100,15 @@ class SelectFromModel(MetaEstimatorMixin, SelectorMixin, BaseEstimator):
         ``threshold`` in the case where the ``coef_`` attribute of the
         estimator is of dimension 2.
 
-    max_features : int, default=None
-        The maximum number of features to select.
+    max_features : int, callable, default=None
+        The maximum number of features to select. ``max_features`` may be a
+        whole number representing the number of maximum features to allow,
+        or a callable that takes only ``X`` as an argument and returns the
+        number of maximum features to allow. When ``max_features`` is a
+        callable, the maximum number of features is calculated as
+        ``max_features_=max_features(X)``. For example, if one wanted to
+        select only up to 50% of the incoming features, they may set
+        ``max_features=lambda X: round(X.shape[1]/2)``.
```

Comment:
This example can be placed in the `Examples` section. You can add a sentence before the code snippet to describe what is happening. For example:

https://github.com/scikit-learn/scikit-learn/blob/581a66316af6b57e6d455b5517c543932f857f0b/sklearn/model_selection/_validation.py#L230-L234